python/lsst/meas/base/applyApCorr.py

```diff
@@ -53,38 +53,52 @@ def __init__(self, schema, name):
+            - {name}_apCorr (only if not already added by proxy)
+            - {name}_apCorrSigma (only if not already added by proxy)
             - {name}_flag_apCorr
-        @param[in] name  field name prefix for flux needing aperture correction, e.g. "base_PsfFlux"
+        @param[in] model  field name prefix for flux with aperture correction model, e.g. "base_PsfFlux"
+        @param[in] name  field name prefix for flux needing aperture correction; may be None if it's the
+            same as for the 'model' parameter
 
         ApCorrInfo has the following attributes:
         - name: field name prefix for flux needing aperture correction
+        - modelName: field name for aperture correction model for flux
+        - modelSigmaName: field name for aperture correction model for fluxSigma
+        - doApCorrColumn: should we write the aperture correction values? (not if they're already being
+             written by a proxy)
         - fluxName: name of flux field
         - fluxSigmaName: name of flux sigma field
         - fluxKey: key to flux field
         - fluxSigmaKey: key to flux sigma field
         - fluxFlagKey: key to flux flag field
-        - apCorrKey: key to new aperture correction field
-        - apCorrSigmaKey: key to new aperture correction sigma field
+        - apCorrKey: key to new aperture correction field (only set if doApCorrColumn=True)
+        - apCorrSigmaKey: key to new aperture correction sigma field (only set if doApCorrColumn=True)
         - apCorrFlagKey: key to new aperture correction flag field
         """
+        if name is None:
+            name = model
         self.name = name
+        self.modelName = model + "_flux"
+        self.modelSigmaName = model + "_fluxSigma"
         self.fluxName = name + "_flux"
         self.fluxSigmaName = name + "_fluxSigma"
         self.fluxKey = schema.find(self.fluxName).key
         self.fluxSigmaKey = schema.find(self.fluxSigmaName).key
         self.fluxFlagKey = schema.find(name + "_flag").key
-        self.apCorrKey = schema.addField(
-            name + "_apCorr",
-            doc="aperture correction applied to %s" % (name,),
-            type=float,
-        )
-        self.apCorrSigmaKey = schema.addField(
-            name + "_apCorrSigma",
-            doc="aperture correction applied to %s" % (name,),
-            type=float,
-        )
+
+        # No need to write the same aperture corrections multiple times
+        self.doApCorrColumn = (name == model or model + "_apCorr" not in schema)
+        if self.doApCorrColumn:
+            self.apCorrKey = schema.addField(
+                name + "_apCorr",
+                doc="aperture correction applied to %s" % (name,),
+                type=float,
+            )
+            self.apCorrSigmaKey = schema.addField(
+                name + "_apCorrSigma",
+                doc="aperture correction applied to %s" % (name,),
+                type=float,
+            )
```

Comment:
if `self.doApCorrColumn` is False, we should create an alias that points to the previously-added column (that's exactly the sort of thing aliases were designed for).